share/lib/python/neuron/rxd/node.py

```diff
@@ -646,13 +647,13 @@ def satisfies(self, condition):
         except:
             pass  # ignore the error and try Node3D specific conditions
 
-        if isinstance(condition, tuple) and len(condition) == 3:
+        if isinstance(condition, tuple) and len(condition) == 3: # Int replaced by floor to accept negative coordinates.
             x, y, z = condition
             mesh = self._r._mesh_grid
             return (
-                int((x - mesh["xlo"]) / mesh["dx"]) == self._i
-                and int((y - mesh["ylo"]) / mesh["dy"]) == self._j
-                and int((z - mesh["zlo"]) / mesh["dz"]) == self._k
+                _floor((x - mesh["xlo"]) / mesh["dx"]) == self._i
```

Comment:
Just curious. The floor is a step function with discontinuity at the integer value.
i.e.
 ```
>>> floor(-.001)
np.float64(-1.0)
>>> floor(0.0)
np.float64(0.0)
>>> floor(.001)
np.float64(0.0)
```
Or did you want f(x) to be 0 for x in ```[-.5, +.5)```
Also, is float to integer comparison ok in this context?

Reply:
The problem is that int rounds to 0, so you end up with the range of (-1, 1) values that mapped to 0. We actually want largest integer no bigger than x.

The implication of this is that the following assert fails if things on the inside go negative with int but passes with floor:

```python
 assert nodes[-1] in nodes((nodes[-1].x3d, nodes[-1].y3d, nodes[-1].z3d))
```

Reply:
The idea here is that integer block 0 runs from position 0 to dx, so any value in that interval should correspond to that block, which reports an x3d of dx/2 ....  (it need not start at 0... it's actually chosen dynamically to be below the entire morphology which is why this was never a problem per se, but now we're trying to combine grids which means things may be off the grid boundaries)